src/Swarm/Doc/Pedagogy.hs

```diff
@@ -159,11 +157,13 @@ generateIntroductionsSequence =
 -- For unit tests, can instead access the scenarios via the GameState.
 loadScenarioCollection :: IO ScenarioCollection
 loadScenarioCollection = simpleErrorHandle $ do
-  entities <- ExceptT loadEntities
-  (failures, loadedScenarios) <- liftIO $ loadScenariosWithWarnings entities
-  when (notNull failures) . liftIO $
-    hPutStrLn stderr "Loading failures: " >> mapM_ (T.hPutStrLn stderr . prettyFailure) failures
-  return loadedScenarios
+  entities <- loadEntities
+  evalAccum (mempty :: Seq SystemFailure) $ loadScenarios entities
+
+-- Note we ignore any warnings generated by 'loadScenarios' above.
+-- Any warnings will be caught when loading all the scenarios via
+-- the usual code path; we do not need to do anything with them
+-- here while simply rendering pedagogy info.
```

Comment:
I think that this comment should be inline with the body of `loadScenarioCollection` (i.e. just above line 161).  As is, it's kind of floating in no-man's land.